File: app/modules/Metrica.lua
local ____exports = {}
local MetricaModule
function MetricaModule()
    local has_metrica = false
    local function init(id_metrica_android)
        if System.platform == "HTML5" then
            if yametrica ~= nil then
                yametrica.not_bounce()
            end
        elseif appmetrica ~= nil then
            has_metrica = id_metrica_android ~= ""
            appmetrica.set_callback(function() return log("app metrica init ok") end)
            if has_metrica then
                appmetrica.initialize(id_metrica_android)
            end
        end
    end
    local function report(event, json_data)
        if json_data == nil then
            json_data = ""
        end
        if appmetrica ~= nil and has_metrica then
            appmetrica.report_event(
                event,
                json_data == "" and "" or json.encode(json_data)
            )
        end
        if yametrica ~= nil then
            yametrica.reach_goal(event, json_data == "" and ({}) or json_data)
        end
    end
    return {init = init, report = report}
end
function ____exports.register_metrica()
    _G.Metrica = MetricaModule()
end
return ____exports

File: src/modules/Metrica.ts
/* eslint-disable @typescript-eslint/no-unsafe-argument */
/* eslint-disable @typescript-eslint/no-unsafe-assignment */
/* eslint-disable @typescript-eslint/no-explicit-any */
/* eslint-disable @typescript-eslint/no-unsafe-member-access */

/*
    Модуль для работы с метрикой
*/

declare global {
    const Metrica: ReturnType<typeof MetricaModule>;
}

export function register_metrica() {
    (_G as any).Metrica = MetricaModule();
}

function MetricaModule() {
    let has_metrica = false;
    
    function init(id_metrica_android: string) {
        if (System.platform == 'HTML5') {
            if (yametrica != null)
                yametrica.not_bounce();
        }
        else if (appmetrica != null) {
            has_metrica = id_metrica_android != '';
            appmetrica.set_callback(() => log('app metrica init ok'));
            if (has_metrica)
                appmetrica.initialize(id_metrica_android);
        }
    }

    function report(event: string, json_data: any = '') {
        if (appmetrica != null && has_metrica)
            appmetrica.report_event(event, json_data == '' ? '' : json.encode(json_data));
        if (yametrica != null)
            yametrica.reach_goal(event, json_data == '' ? {} : json_data);
    }

    return { init, report };
}